bot/handlers/role_selection.go

```diff
@@ -92,97 +86,115 @@ func (r *RoleSelectionHandler) ReconcileConfig(s *discordgo.Session) error {
 					continue
 				}
 			}
+		} else {
+			r.prompts[existingMessage.ID] = p
+			for _, op := range p.Options {
+				var reactionExists bool
+				for _, reaction := range existingMessage.Reactions {
+					if reaction.Emoji.ID == op.EmojiID {
+						reactionExists = true
+						break
+					}
+				}
+				if !reactionExists {
+					err := s.MessageReactionAdd(existingMessage.ChannelID, existingMessage.ID, op.EmojiID)
+					if err != nil {
+						r.logger.Error("error adding reaction",
+							zap.String("channelName", roleChannel.Name),
+							zap.String("channelId", roleChannel.ID),
+							zap.String("emojiID", op.EmojiID),
+							zap.String("option", op.Description),
+							zap.Error(err))
+						continue
+					}
+				}
+			}
 		}
 	}
+	r.initialized = true
 	return nil
 }
 
-// HandleReactionAdd handles a reaction add event.
-func (r *RoleSelectionHandler) HandleReactionAdd(s *discordgo.Session, m *discordgo.MessageReactionAdd) {
+// handleReaction is a helper function to handle both reaction add and remove events.
```

Comment:
Add a meaningful description that explains what each return value represents @jules